openadapt/window/_macos.py

```diff
@@ -135,14 +142,58 @@ def dump_state(element, elements=None):
 def deepconvert_objc(object):
     """Convert all contents of an ObjC object to Python primitives."""
     value = object
+    strings = (
+        str,
+        AppKit.NSString,
+        ApplicationServices.AXTextMarkerRangeRef,
+        ApplicationServices.AXUIElementRef,
+        ApplicationServices.AXTextMarkerRef,
+        Quartz.CGPathRef,
+    )
+
     if isinstance(object, AppKit.NSNumber):
         value = int(object)
     elif isinstance(object, AppKit.NSArray) or isinstance(object, list):
         value = [deepconvert_objc(x) for x in object]
     elif isinstance(object, AppKit.NSDictionary) or isinstance(object, dict):
-        value = dict(object)
-        for k, v in value.items():
-            value[k] = deepconvert_objc(v)
+        value = {deepconvert_objc(k): deepconvert_objc(v) for k, v in object.items()}
+    elif isinstance(object, strings):
+        value = str(object)
+    # handle core-foundation class AXValueRef
+    elif isinstance(object, ApplicationServices.AXValueRef):
+        # convert to dict - note: this object is not iterable
+        rep = repr(object)
+        x_value = re.search(r"x:([\d.]+)", rep)
+        y_value = re.search(r"y:([\d.]+)", rep)
+        w_value = re.search(r"w:([\d.]+)", rep)
+        h_value = re.search(r"h:([\d.]+)", rep)
+        type_value = re.search(r"type\s?=\s?(\w+)", rep)
+        value = {
+            "x": float(x_value.group(1)) if x_value else None,
+            "y": float(y_value.group(1)) if y_value else None,
+            "w": float(w_value.group(1)) if w_value else None,
+            "h": float(h_value.group(1)) if h_value else None,
+            "type": type_value.group(1) if type_value else None,
+        }
+    elif isinstance(object, Foundation.NSURL):
+        value = str(object.absoluteString())
+    elif isinstance(object, Foundation.__NSCFAttributedString):
+        value = str(object.string())
+    elif isinstance(object, Foundation.__NSCFData):
+        value = {
+            deepconvert_objc(k): deepconvert_objc(v)
+            for k, v in plistlib.loads(object).items()
+        }
+    elif isinstance(object, plistlib.UID):
+        value = object.data
+    else:
+        if not (isinstance(object, bool) or isinstance(object, int)):
```

Comment:
Please also check for `None`

Reply:
I've added a check to ignore None [8eb7194](https://github.com/OpenAdaptAI/OpenAdapt/pull/323/commits/8eb7194f3e17aa0f2f8835622d14eb5c02b197d8)